src/image-featurizer/src/main/scala/ImageAugmenter.scala

```diff
@@ -0,0 +1,73 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in project root for information.
+
+package com.microsoft.ml.spark
+
+import com.microsoft.ml.spark.schema.ImageSchema
+import org.apache.spark.ml._
+import org.apache.spark.ml.param._
+import org.apache.spark.ml.util.{DefaultParamsReadable, DefaultParamsWritable, Identifiable}
+import org.apache.spark.sql._
+import org.apache.spark.sql.types._
+
+object ImageAugmenter extends DefaultParamsReadable[ImageAugmenter]
+
+class ImageAugmenter(val uid: String) extends Transformer
+  with HasInputCol with HasOutputCol with DefaultParamsWritable {
+
+  def this() = this(Identifiable.randomUID("ImageAugmenter"))
+
+  // if the problem is symmetric, the training dataset is supplemented with the "flipped" images
+  // by default, we assume the left-right symmetry, but not up-down
+  val flipLeftRight = BooleanParam(this, "flipLeftRight", "symmetric lr", false)
+
```

Comment:
OK.. I would suggest structuring this similar to ImageTransformer where you can use fluent syntax to build the pipeline of transformations. 

Also, please add some parameters that are just not boolean (affine transforms for example).